src/lib/layouts/DocsTutorial.svelte

```diff
@@ -3,77 +3,103 @@
     import { Feedback } from '$lib/components';
     import type { Tutorial } from '$markdoc/layouts/Tutorial.svelte';
     import type { TocItem } from './DocsArticle.svelte';
+    import Heading from '$markdoc/nodes/Heading.svelte';
+    import { onMount } from 'svelte';
 
-    export let title: string;
     export let toc: Array<TocItem>;
     export let currentStep: number;
-    export let back: string;
     export let date: string;
 
     export let tutorials: Array<Tutorial>;
 
+    const firstStepItem = tutorials[0];
+    // currentStep starts from 1, the arrays start from 0.
+    const currentStepItem = (tutorials[currentStep - 1] ?? firstStepItem);
+
     $: nextStep = tutorials.find((tutorial) => tutorial.step === currentStep + 1);
     $: prevStep = tutorials.find((tutorial) => tutorial.step === currentStep - 1);
+
+    // `any` for compatibility with reactive variables.
+    function getCorrectTitle(tutorial: any | Tutorial, checkAt: number): string {
```

Comment:
But it's always a Tutorial right? I would prefer to keep this one strict, and the cast the variable that is thrown into the function when needed.

Reply:
Aah yes, it is. I think I added `any` because `prevStep` returns `any|Tutorial`, not sure why. The type should be available there.